### PR TITLE
Force logout when session token expires

### PR DESCRIPTION
**Pull request by vicmgs**
currently, local auth0 sessions will expire in 30 seconds so you can test this code on local for now. I'll revert the timer back to 10 hours once PR is merged.

### CODE REVIEW

File: assets/js/actions/apollo.js [hunk 1]
```diff
@@ -102,27 +103,20 @@ export const setupApolloClient = (getAuthToken, organizationId) => {
     socket.connect()
 
     store.subscribe(async () => {
-      const newTokenClaims = await getAuthToken();
-      const newToken = newTokenClaims.__raw;
-      const newOrganization = store.getState().organization;
-
-      if (newToken !== token || currentOrganizationId !== newOrganization.currentOrganizationId) {
```
Comment (vicmgs) [inline assets/js/actions/apollo.js:109]: none of this ever does anything, switch orgs reloads the page entirely, bypassing this

File: assets/js/util/rest.js [hunk 2]
```diff
@@ -51,9 +50,15 @@ const headers = async () => {
   if (organizationId) {
     Object.assign(headerParams, { organization: organizationId })
   }
-  const tokenClaims = await getIdTokenClaims();
```
Comment (vicmgs) [inline assets/js/util/rest.js:54]: this generate a new token everytime, and thus never expires. This is why in the future state, I only use it for the first rest api call and use the stored token in global state once apolloClient has been created so we can keep using the same token.

File: assets/js/actions/apollo.js [hunk 3]
```diff
@@ -102,27 +103,20 @@ export const setupApolloClient = (getAuthToken, organizationId) => {
...
-        currentOrganizationId = newOrganizationId
-        token = newToken;
-        socket.disconnect();
-        socket = createSocket(token, currentOrganizationId)
-        socket.connect()
+      if (Math.ceil(Date.now() / 1000) > store.getState().apollo.tokenClaims.exp) {
+        logout()
```
Comment (ambuznego) [inline assets/js/actions/apollo.js:107]: could we alert user before this happens and give them an opportunity to renew the session?

Comment (vicmgs) [reply assets/js/actions/apollo.js:107]: i still havent been able to figure out token refresh, so probably try to make that in a separate PR. In the meantime, I'll make the token expiration on prod like 36 hours or something

**Comment by ambuznego** [comment]
seems like the session still expires if user is active, is that something we can address w/ auth0 to extend the session?

File: assets/js/actions/apollo.js (hunk 3 above)
Comment (vicmgs) [reply assets/js/actions/apollo.js:107]: the docs are terrible

Comment (ambuznego) [reply assets/js/actions/apollo.js:107]: for sure, we can address the immediate urgency of logging user out and tackle refreshing session in future ticket to improve UX

**Review by ambuznego — APPROVED** [review]

**Pull request merged**